test/Microsoft.DotNet.Docker.Tests/ImageTests.cs

```diff
@@ -0,0 +1,179 @@
+
+            try
+            {
+                string dotNetCmd = $"dotnet publish -o {DockerHelper.ContainerWorkDir}";
+                DockerHelper.Run(appSdkImage, dotNetCmd, frameworkDepAppId, frameworkDepAppId);
+
+                string runtimeImage = GetDotNetImage(runtimeImageVersion, DotNetImageType.Runtime);
+                string appDllPath = DockerHelper.GetContainerWorkPath("test.dll");
+                DockerHelper.Run(runtimeImage, $"dotnet {appDllPath}", frameworkDepAppId, frameworkDepAppId);
+            }
+            finally
+            {
+                DockerHelper.DeleteVolume(frameworkDepAppId);
+            }
+        }
+
+        private void VerifyRuntimeDepsImage_SelfContainedApp(
+            string runtimeDepsImageVersion, string appSdkImage, bool usePreReleasePackageFeed)
+        {
+            string selfContainedAppId = GetIdentifier(runtimeDepsImageVersion, "self-contained-app");
+            string rid = "debian.8-x64";
+
+            try
+            {
+                List<string> additionalArgs = new List<string>();
+                additionalArgs.Add($"rid={rid}");
+                if (runtimeDepsImageVersion == "2.0")
+                {
+                    additionalArgs.Add($"optional_restore_args=/p:runtimeidentifier={rid}");
+                }
+
+                string buildArgs = GetBuildArgs(usePreReleasePackageFeed, additionalArgs.ToArray());
+                DockerHelper.Build("Dockerfile.linux.publish", appSdkImage, selfContainedAppId, buildArgs);
+
+                try
+                {
+                    string optionalPublishArgs = runtimeDepsImageVersion.StartsWith("1.") ? "" : "--no-restore";
+                    string dotNetCmd = $"dotnet publish -r {rid} -o {DockerHelper.ContainerWorkDir} {optionalPublishArgs}";
+                    DockerHelper.Run(selfContainedAppId, dotNetCmd, selfContainedAppId, selfContainedAppId);
+
+                    string runtimeDepsImage = GetDotNetImage(runtimeDepsImageVersion, DotNetImageType.Runtime_Deps);
+                    string appExePath = DockerHelper.GetContainerWorkPath("test");
+                    DockerHelper.Run(runtimeDepsImage, appExePath, selfContainedAppId, selfContainedAppId);
+                }
+                finally
+                {
+                    DockerHelper.DeleteVolume(selfContainedAppId);
+                }
+            }
+            finally
+            {
+                DockerHelper.DeleteImage(selfContainedAppId);
+            }
+        }
+
+        private static string GetBuildArgs(bool usePreReleasePackageFeed, params string[] additionalArgs)
+        {
+            string buildArgs = string.Empty;
+
+            if (usePreReleasePackageFeed)
+            {
+                buildArgs += " --build-arg optional_restore_feeds=\"-s https://dotnet.myget.org/F/dotnet-core/api/v3/index.json -s https://api.nuget.org/v3/index.json\"";
```

Comment:
Minor - move these constants to top of this file or a separate constants file.

Reply:
hmm - what value would that provide.

Reply:
Maintainability.  For example, if I were to switch from v3 to vNext feeds, then I would look for a variable or a property corresponding to the feed. Having such variables clustered at the top would make it easier to locate and edit.

I know we can argue that in the current situation, the feeds are defined-used in only one line/statement, and thus a need for a variable is not required. So in this case we can keep it the way it is.